File: objectModel/CSharp/Microsoft.CommonDataModel.ObjectModel/Persistence/CdmFolder/Types/DocumentContent.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

namespace Microsoft.CommonDataModel.ObjectModel.Persistence.CdmFolder.Types
{
    using Newtonsoft.Json;
    using Newtonsoft.Json.Linq;
    using System.Collections.Generic;
    public class DocumentContent
    {
        [JsonProperty("jsonSchemaSemanticVersion")]
        public string JsonSchemaSemanticVersion { get; set; }

        [JsonProperty("$schema", NullValueHandling = NullValueHandling.Ignore)]
        public string Schema { get; set; }

        [JsonProperty("imports", NullValueHandling = NullValueHandling.Ignore)]
        public List<Import> Imports { get; set; }

        [JsonProperty("definitions", NullValueHandling = NullValueHandling.Ignore)]
        public List<JToken> Definitions { get; set; }

        [JsonProperty("documentVersion", NullValueHandling = NullValueHandling.Ignore)]
        public string DocumentVersion { get; set; }
    }
}

File: objectModel/Java/objectmodel/src/main/java/com/microsoft/commondatamodel/objectmodel/persistence/cdmfolder/types/DocumentContent.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel.persistence.cdmfolder.types;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.JsonNode;
import java.util.List;

public class DocumentContent {
    private String jsonSchemaSemanticVersion;

    @JsonProperty("$schema")
    @JsonInclude(JsonInclude.Include.NON_NULL)
    private String schema;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    private List<Import> imports;

    @JsonInclude(JsonInclude.Include.NON_NULL)
    private List<JsonNode> definitions;

    public String getJsonSchemaSemanticVersion() {
        return jsonSchemaSemanticVersion;
    }

    public void setJsonSchemaSemanticVersion(final String jsonSchemaSemanticVersion) {
        this.jsonSchemaSemanticVersion = jsonSchemaSemanticVersion;
    }

    public String getSchema() {
        return schema;
    }

    public void setSchema(final String schema) {
        this.schema = schema;
    }

    public List<Import> getImports() {
        return imports;
    }

    public void setImports(final List<Import> imports) {
        this.imports = imports;
    }

    public List<JsonNode> getDefinitions() {
        return definitions;
    }

    public void setDefinitions(List<JsonNode> definitions) {
        this.definitions = definitions;
    }
}
